crates/oxc_linter/src/rules/unicorn/consistent_existence_index_check.rs

```diff
@@ -0,0 +1,493 @@
+
+impl Rule for ConsistentExistenceIndexCheck {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::BinaryExpression(binary_expression) = node.kind() else {
+            return;
+        };
+
+        let left = binary_expression.left.get_inner_expression();
+        let right = binary_expression.right.get_inner_expression();
+        let operator = binary_expression.operator;
+
+        let Expression::Identifier(identifier) = left else {
+            return;
+        };
+
+        let Some(reference_id) = identifier.reference_id.get() else {
+            return;
+        };
+
+        let Some(reference_symbol_id) = ctx.symbols().get_reference(reference_id).symbol_id()
+        else {
+            return;
+        };
+
+        let declaration = ctx.symbols().get_declaration(reference_symbol_id);
+        let node = ctx.nodes().get_node(declaration);
+
+        if let AstKind::VariableDeclarator(variables_declarator) = node.kind() {
+            if variables_declarator.kind != VariableDeclarationKind::Const {
+                return;
+            }
+
+            let Some(Expression::CallExpression(call)) = &variables_declarator.init else {
+                return;
+            };
+
+            if !call.callee.is_member_expression() {
+                return;
+            }
+
+            let Some(callee_name) = call.callee_name() else {
+                return;
+            };
+
+            if !METHOD_NAMES.contains(&callee_name) {
+                return;
+            }
+
+            let replacement = get_replacement(right, &operator);
+
+            let Some(replacement) = &replacement else {
+                return;
+            };
+
+            let existence_or_non_existence =
+                if replacement.replacement_value == "-1" { "non-existence" } else { "existence" };
+
+            let label = format!(
+                        "Prefer `{replacement_operator} {replacement_value}` over `{original_operator} {original_value}` to check {existenceOrNonExistence}.",
+                        replacement_operator = replacement.replacement_operator,
+                        replacement_value = replacement.replacement_value,
+                        original_operator = replacement.original_operator,
+                        original_value = replacement.original_value,
+                        existenceOrNonExistence = existence_or_non_existence,
+                    );
+
+            let operator_start = binary_expression.left.span().end;
+            let operator_end = binary_expression.right.span().start;
+            let operator_span = Span::new(operator_start, operator_end);
+            let operator_source = ctx.source_range(operator_span);
+
+            let operator_matches = operator_source.match_indices(replacement.original_operator);
+            let mut operator_replacement_text = operator_source.to_string();
```

Comment:
Let's move all of the fix generation inside of the fixer closure. That way we don't need to run it unless fixes are enabled.